react-catalog/my-app/src/Catalog.tsx

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useState } from 'react';
+import { ProductCard } from './ProductCard';
+import { Product, readCatalog } from './lib';
+
+export function Catalog() {
+  const [products, setProducts] = useState<Product[]>();
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    async function getProducts() {
+      try {
+        const products = await readCatalog();
+        setProducts(products);
+      } catch (err) {
+        console.log(err);
```

Comment:
You need to somehow let the user know that an error occurred. You can do this by displaying the error on the page or with `alert()`. But if you just log it the user will never know that an error occurred and will be upset with you!